arrow-array/src/array/fixed_size_binary_array.rs

```diff
@@ -129,17 +129,29 @@ impl FixedSizeBinaryArray {
     /// # Errors
     ///
     /// Returns error if argument has length zero, or sizes of nested slices don't match.
-    pub fn try_from_sparse_iter<T, U>(mut iter: T) -> Result<Self, ArrowError>
+    pub fn try_from_sparse_iter<T, U>(iter: T) -> Result<Self, ArrowError>
+    where
+        T: Iterator<Item = Option<U>>,
+        U: AsRef<[u8]>,
+    {
+        Self::try_from_sparse_iter_with_size(iter, None)
+    }
+
+    pub fn try_from_sparse_iter_with_size<T, U>(
```

Comment:
```suggestion
    pub fn try_from_sparse_iter_with_size<T, U>(iter: T, size: i32) -> Result<Self, ArrowError> {
        try_from_sparse_iter_with_size_opt::<T, U>(iter, Some(size))
    }
    
    fn try_from_sparse_iter_with_size_opt<T, U>(
```

I think this makes for better ergonomics

Reply:
I made the parameter Optional instead